### PR TITLE
feat: add projects api

### PR DESCRIPTION
**Pull request by krtkvrm**
- [x] Organization API
     - [x] Create
     - [x] List
     - [x] Get
     - [x] Update
- [x]  PG schema Migrations & Rollbacks
- [x] Refactor Org Service, to remove redundant `*Organization`
- [x] Remove `version` column in org table
- [x] Apply PR Reviews

### CODE REVIEW

File: internal/project/project.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
+type Service struct {
+	Store Store
+}
+
+var (
+	ProjectDoesntExist = errors.New("org doesn't exist")
```
Comment (AkarshSatija) [inline internal/project/project.go:23]: Typo at error message

File: api/handler/v1/project.go [hunk 2]
```diff
@@ -2,25 +2,157 @@ package v1
...
+
+	metaData, err := structpb.NewStruct(mapOfInterfaceValues(newProject.Metadata))
+	if err != nil {
+		logger.Error(err.Error())
+		return nil, grpcInternalServerError
+	}
+
+	return &shieldv1.CreateProjectResponse{Project: &shieldv1.Project{
+		Id:        newProject.Id,
+		Name:      newProject.Name,
```
Comment (AkarshSatija) [inline api/handler/v1/project.go:83]: Where are we keeping project <> org mapping?

File: api/handler/v1/project.go [hunk 3]
```diff
@@ -2,25 +2,157 @@ package v1
...
 func (v Dep) CreateProject(ctx context.Context, request *shieldv1.CreateProjectRequest) (*shieldv1.CreateProjectResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method not implemented")
+	logger := grpczap.Extract(ctx)
+	metaDataMap, err := mapOfStringValues(request.GetBody().Metadata.AsMap())
+	if err != nil {
+		logger.Error(err.Error())
+		return nil, grpcBadBodyError
+	}
+
+	slug := request.GetBody().Slug
```
Comment (AkarshSatija) [inline api/handler/v1/project.go:59]: Would need validations as well that slug can't contain spaces or special chars except for hyphens and underscore. Also, let's do a lowercase() on it.

File: api/handler/v1/project.go (hunk 2 above)
Comment (krtkvrm) [reply api/handler/v1/project.go:83]: yes, will make changes here and as well as in proton
https://github.com/odpf/proton/pull/56

Comment (krtkvrm) [reply api/handler/v1/project.go:84]: changes are incorporated

**Comment by krtkvrm** [comment]
The scope of this PR is also to refactor Org Service. 
from:
```go
type Store interface {
	GetOrganization(ctx context.Context, id string) (org.Organization, error)
	CreateOrganization(ctx context.Context, org org.Organization) (org.Organization, error)
	ListOrganizations(ctx context.Context) ([]org.Organization, error)
	UpdateOrganization(ctx context.Context, toUpdate org.Organization) (org.Organization, error)
}
```

To:
```go
type OrganizationService interface {
	Get(ctx context.Context, id string) (modelv1.Organization, error)
	Create(ctx context.Context, org modelv1.Organization) (modelv1.Organization, error)
	List(ctx context.Context) ([]modelv1.Organization, error)
	Update(ctx context.Context, toUpdate modelv1.Organization) (modelv1.Organization, error)
}
```

File: api/handler/v1/org.go [hunk 4]
```diff
@@ -18,10 +19,10 @@ import (
 )
 
 type OrganizationService interface {
-	GetOrganization(ctx context.Context, id string) (org.Organization, error)
-	CreateOrganization(ctx context.Context, org org.Organization) (org.Organization, error)
-	ListOrganizations(ctx context.Context) ([]org.Organization, error)
-	UpdateOrganization(ctx context.Context, toUpdate org.Organization) (org.Organization, error)
+	GetOrganization(ctx context.Context, id string) (modelv1.Organization, error)
```
Comment (ravisuhag) [inline api/handler/v1/org.go]: @krtkvrm I think we don't need to version internal models. So would suggest removing versioning from here.

Comment (krtkvrm) [reply api/handler/v1/org.go]: yeah.. makes sense 👍

Comment (krtkvrm) [reply api/handler/v1/org.go]: done

**Comment by krtkvrm** [comment]
There are many checkpoint commits, so please when merging please use `squash and merge`

**Comment by krtkvrm** [comment]
There are many checkpoint commits, so please when merging please use `squash and merge`

File: api/handler/v1/project.go [hunk 5]
```diff
@@ -2,25 +2,160 @@ package v1
...
+	if strings.TrimSpace(slug) == "" {
+		slug = generateSlug(request.GetBody().Name)
+	}
+
+	newProject, err := v.ProjectService.Create(ctx, model.Project{
+		Name:     request.GetBody().Name,
+		Slug:     slug,
+		Metadata: metaDataMap,
+		//Organization: org.Organization{Id: "ACCEPT"},
```
Comment (krtkvrm) [inline api/handler/v1/project.go:70]: Once odpf/proton#56 is merged, this would be updated

Comment (ravisuhag) [reply api/handler/v1/project.go:70]: Approved and merged.

**Review by AkarshSatija — APPROVED** [review]

**Pull request merged**